client/src/components/AuthModal.jsx

```diff
@@ -0,0 +1,282 @@
+    backgroundColor: "#f2f2f2",
+  },
+  textField: {
+    borderRadius: `4px 4px 4px 4px`,
+    outline: "none",
+    padding: "7px",
+    width: "250px",
+    backgroundColor: "white",
+  },
+  textFieldInput: {
+    width: "calc(100% - 24px)",
+  },
+  typography: {
+    textAlign: "center",
+    display: "block",
+  },
+  box: {
+    marginBottom: "20px",
+  },
+  h4: {
+    textAlign: "center",
+    margin: "20px 0 20px 0",
+  },
+  button: {
+    height: 56,
+    borderRadius: 28,
+    width: 142,
+    display: "block",
+    margin: "auto",
+    marginBottom: "20px",
+  },
+  footer: {
+    textAlign: "center",
+    padding: "20px 20px 20px 20px ",
+  },
+  hyperlink: {
+    display: "inline",
+    color: "#DF1B1B",
+    fontWeight: "bold",
+    cursor: "pointer",
+  },
+}));
+
+const AuthModal = ({ isAuthPage }) => {
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [signedIn, setSignedIn] = useState(true);
+  const [isFirst, setIsFirst] = useState(true);
+
+  const history = useHistory();
+
+  useEffect(async () => {
+    setIsFirst(false);
+    let results = await fetch("/user/getCurrentUser");
+
+    results = await results.json();
+
+    if (results.user === undefined) {
+      setSignedIn(false);
+    } else {
+      setSignedIn(true);
+      history.push("/dashboard");
+    }
+  }, []);
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+
+    if (password.length < 7) {
+      alert("Password must be greater than 6");
+      return;
+    }
+
+    if (name.trim().length === 0 && pathname === "/signup") {
+      alert("Name must not be just spaces");
+      return;
+    }
+
+    try {
+      //email is handled by the TextField
+      if (pathname === "/signup") {
+        let result = await fetch("/user/register", {
+          method: "POST",
+          headers: { "Content-type": "application/json" },
+          body: JSON.stringify({
+            name,
+            email,
+            password,
+          }),
+        });
+
+        result = await result.json();
+
+        setSignedIn(
+          result.user !== undefined && result.user.id !== undefined
+            ? true
+            : false
+        );
+      } else {
+        let result = await fetch("/user/login", {
+          method: "POST",
+          headers: { "Content-type": "application/json" },
+          body: JSON.stringify({
+            email,
+            password,
+          }),
+        });
+
+        result = await result.json();
+
+        setSignedIn(
+          result.user !== undefined && result.user.id !== undefined
+            ? true
+            : false
+        );
+      }
+      history.push("/dashboard");
+    } catch (error) {
+      alert("Something went wrong");
+      console.error(error);
+    }
+  };
+
+  const classes = useStyles();
+  const { pathname } = window.location;
+
+  return (
+    <>
+      {signedIn  ? (
+        <></>
```

Comment:
If you don't want to display anything if the user is signed in, you shouldn't use a `Fragment` (`<></>`). These are only used for when you want to return more than one JSX component and you don't want to use an array to do that. 
If you want to return nothing, what's usually used is `null`.